### DISCUSSION TITLE
Customize the 'Draw a track map with numbered corners' code

### DISCUSSION THREAD

**Discussion by ghost** [theOehrly/Fast-F1#499] [Q&A]
Hello everyone, 
For my research project related to porpoising in Formula 1, I need to draw F1 circuits while highlighting the straight lines in a different color in order to shed light on this phenomena that happens especially on high-speed straight lines. I stumbled upon the python package but I don't know what to change in order to draw F1 circuits as I expect. 
Can someone explain to me how can I do it ?
Thank you in advance for any feedback
Happy new year to everybode
admz44

**Comment by theOehrly** [comment]
Hi, have you coded in Python (or another language) before?
In general, you want to do something like in this example: https://docs.fastf1.dev/examples_gallery/plot_gear_shifts_on_track.html#sphx-glr-examples-gallery-plot-gear-shifts-on-track-py
Of course, you need to replace the selected gear with a metric that indicates whether you are on a straight or not. You could do that manually for each circuit. Or you could calculate the curvature of the trajectory to do it in an automated way. There's no ready-made solution here, so you will need to play around a bit.

**Reply by Casper-Guo** [reply]
Something like the following might work:

1, sample the location data every 0.5 seconds (this can be adjusted to see what works best), for every two samples calculate the slope between them, and then you can use the change in slopes to categorize the shape of the circuit. Very long winded way to say calculate the second derivative of the circuit. I am sure numerical libraries have efficient methods for this

2, A more quick and easy solution is just inferring this from brake/throttle
